gravann/_train_v2.py

```diff
@@ -0,0 +1,228 @@
+import pickle as pk
+import time
+from collections import deque
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from tqdm import tqdm
+
+from . import plot_model_rejection
+from ._encodings import *
+from ._io import save_results, save_plots_v2
+from ._losses import contrastive_loss, normalized_relative_L2_loss, normalized_relative_component_loss
+# Required for loading runs
+from ._train_v2_init import init_training_sampler, init_environment, init_model_and_optimizer, init_prediction_label, \
+    init_ground_truth_labels, init_input_data, init_noise
+from ._validation import validation_results_unpack_df
+from ._validation_v2 import validation_v2
+
+
+def train_on_batch_v2(points, prediction_fn, labels, loss_fn, optimizer, scheduler):
```

Comment:
v2? Please ideally just replace the originals that I can see what changed otherwise, this is a lot of code to review

Reply:
👍 Until now, I mostly stuck to adding `v2` methods not to break e.g., the jupyter notebooks in the root folder. The refactoring would take a considerable amount of time. But if it's okay to break compatibility, I'll go for it. This will drastically reduce the file sizes 😀